app/src/main/java/org/mozilla/fenix/tabtray/TabTrayFragment.kt

```diff
@@ -192,6 +205,9 @@ class TabTrayFragment : Fragment(R.layout.fragment_tab_tray), TabsTray.Observer,
 
         if (navController.currentDestination?.id == R.id.collectionCreationFragment) return
 
+        // Only register the observer right before moving to collection creation
+        registerCollectionStorageObserver()
```

Comment:
In HomeFragment we have an `unregister` in `onStart`-- should we be doing that here?  Adding an `unregister` in on `onStart` feels really weird.

Reply:
I agree. I'm not sure if that actually helps :/